br/pkg/storage/azblob.go

```diff
@@ -472,18 +472,31 @@ func (s *AzureBlobStorage) DeleteFiles(ctx context.Context, names []string) erro
 }
 
 // Open implements the StorageReader interface.
-func (s *AzureBlobStorage) Open(ctx context.Context, name string) (ExternalFileReader, error) {
+func (s *AzureBlobStorage) Open(ctx context.Context, name string, o *ReaderOption) (ExternalFileReader, error) {
 	client := s.containerClient.NewBlockBlobClient(s.withPrefix(name))
 	resp, err := client.GetProperties(ctx, nil)
 	if err != nil {
 		return nil, errors.Annotate(err, "Failed to get properties from the azure blob")
 	}
 
+	pos := int64(0)
+	totalSize := *resp.ContentLength
+	endPos := totalSize
+	if o != nil {
```

Comment:
didn't check the validity of start/end offset

Reply:
leave it for future. currently I don't know the expected behaviour because it's not widely used, maybe we need to unify the behaviour of different implement after we learn it a lot. (for example, should we return EOF or some errors if start>end for all storage backend)